opentelemetry-sdk/src/trace/provider.rs

```diff
@@ -275,50 +278,73 @@ mod tests {
 
     #[test]
     fn test_tracer_provider_default_resource() {
-        // If users didn't provide a resource and there isn't a env var set. Use default one.
-        let assert_service_name = |provider: super::TracerProvider,
-                                   expect: Option<&'static str>| {
+        let assert_resource = |provider: &super::TracerProvider,
+                               resource_key: &'static str,
+                               expect: Option<&'static str>| {
             assert_eq!(
                 provider
                     .config()
                     .resource
-                    .get(Key::from_static_str("service.name"))
+                    .get(Key::from_static_str(resource_key))
                     .map(|v| v.to_string()),
                 expect.map(|s| s.to_string())
             );
         };
+        let assert_telemetry_resource = |provider: &super::TracerProvider| {
+            assert_eq!(
+                provider
+                    .config()
+                    .resource
+                    .get(TELEMETRY_SDK_LANGUAGE.into()),
+                Some(Value::from("rust"))
+            );
+            assert_eq!(
+                provider.config().resource.get(TELEMETRY_SDK_NAME.into()),
+                Some(Value::from("opentelemetry"))
+            );
+            assert_eq!(
+                provider.config().resource.get(TELEMETRY_SDK_VERSION.into()),
+                Some(Value::from(env!("CARGO_PKG_VERSION")))
+            );
+        };
```

Comment:
Good to investigate (probably as a separate PR) if we can reuse the above validation function - `assert_resource()` and `assert_telemetry_resource()` across all the providers. One option could be to move them to [testing](https://github.com/open-telemetry/opentelemetry-rust/tree/main/opentelemetry-sdk/src/testing) directory, and they can take `resource` object instead of `provider` as an argument.